polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -1318,6 +1367,61 @@ where
 	Ok(())
 }
 
+// Starts a worker thread that runs the approval voting subsystem.
+pub async fn start_approval_worker<
+	WorkProvider: ApprovalVotingWorkProvider + Send + 'static,
+	Sender: SubsystemSender<ChainApiMessage>
+		+ SubsystemSender<RuntimeApiMessage>
+		+ SubsystemSender<ChainSelectionMessage>
+		+ SubsystemSender<AvailabilityRecoveryMessage>
+		+ SubsystemSender<DisputeCoordinatorMessage>
+		+ SubsystemSender<CandidateValidationMessage>
+		+ Clone,
+	ADSender: SubsystemSender<ApprovalDistributionMessage>,
+>(
+	work_provider: WorkProvider,
+	to_other_subsystems: Sender,
+	to_approval_distr: ADSender,
+	config: Config,
+	db: Arc<dyn Database>,
+	keystore: Arc<LocalKeystore>,
+	sync_oracle: Box<dyn SyncOracle + Send>,
+	metrics: Metrics,
+	spawner: Arc<dyn overseer::gen::Spawner + 'static>,
+	clock: Arc<dyn Clock + Send + Sync>,
+) -> SubsystemResult<()> {
+	let approval_voting = ApprovalVotingSubsystem::with_config_and_clock(
+		config,
+		db.clone(),
+		keystore,
+		sync_oracle,
+		metrics,
+		clock,
+		spawner,
+	);
+	let backend = DbBackend::new(db.clone(), approval_voting.db_config);
+	let spawner = approval_voting.spawner.clone();
+	spawner.spawn_blocking(
+		"approval-voting-rewrite-db",
+		Some("approval-voting-rewrite-subsystem"),
```

Comment:
why is it called rewrite db/subsystem? as opposed to just approval-voting-subsystem?

Reply:
This function is actually called and instantiated by the approval-voting-parallel in subsequent PR's, this name was just a placeholder, but the right approach is to not hardcode this here and allow the caller to pass in any name they want, which I did here: https://github.com/paritytech/polkadot-sdk/pull/4846/commits/377f4e1bcd1434a5dc3ecaef2f554afa2e9f168c